code/AssetLib/glTF2/glTF2Asset.inl

```diff
@@ -980,18 +981,20 @@ void Accessor::ExtractData(T *&outData) {
     }
 
     const size_t maxSize = GetMaxByteSize();
-    if (count * stride > maxSize) {
-        throw DeadlyImportError("GLTF: count*stride ", (count * stride), " > maxSize ", maxSize, " in ", getContextForErrorMessages(id, name));
+    if (usedCount * stride > maxSize) {
+        throw DeadlyImportError("GLTF: count*stride ", (usedCount * stride), " > maxSize ", maxSize, " in ", getContextForErrorMessages(id, name));
     }
 
-    outData = new T[count];
-    if (stride == elemSize && targetElemSize == elemSize) {
+    outData = new T[usedCount];
+    if (remappingIndices == nullptr && stride == elemSize && targetElemSize == elemSize) {
         memcpy(outData, data, totalSize);
     } else {
-        for (size_t i = 0; i < count; ++i) {
-            memcpy(outData + i, data + i * stride, elemSize);
+        for (size_t i = 0; i < usedCount; ++i) {
+            size_t srcIdx = remappingIndices != nullptr ? static_cast<size_t>((*remappingIndices)[i]) : i;
+            memcpy(outData + i, data + srcIdx * stride, elemSize);
```

Comment:
I considered breaking this up into two loops, however on a PC the CPU's branch prediction works so well that you will probably not notice any difference in timings. I thought it would be more worthwhile to keep the code compact.

Reply:
Edit: Now that I realized that I have to do separate index range checking in the index case, I will break it up into two loops